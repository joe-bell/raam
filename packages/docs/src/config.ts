import manifest from "../public/manifest.json";

const domain = "raam.joebell.co.uk";

const social = {
  github: "joe-bell",
  twitter: "joebell_",
};

const config = {
  navigation: [
    {
      title: "Getting Started",
      url: "#hooks",
    },
    {
      title: "Recipes",
      url: "#recipes",
    },
    {
      title: "GitHub",
      url: `https://github.com/${social.github}/raam`,
      external: true,
    },
    {
      title: "Twitter",
      url: `https://twitter.com/${social.twitter}`,
      external: true,
    },
  ],
  meta: {
    title: manifest.name,
    description: manifest.description,
    author: "Joe Bell",
    domain,
    url: `https://${domain}`,
    social,
  },
};

export default config;
