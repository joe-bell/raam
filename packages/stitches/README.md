# @raam/stitches

[See the Docs](https://raam.joebell.co.uk)
