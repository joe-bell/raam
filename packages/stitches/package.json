{
  "name": "@raam/stitches",
  "version": "0.2.3",
  "description": "Beautifully boring cosmetic-free Stitches utils for structure and layout.",
  "source": "src/index.ts",
  "main": "dist/index.js",
  "module": "dist/index.esm.js",
  "typings": "dist/index.d.ts",
  "files": [
    "dist",
    "README.md"
  ],
  "scripts": {
    "build": "microbundle --no-compress --jsx React.createElement --tsconfig tsconfig.json",
    "clean": "rimraf {dist,.rts2*}",
    "lint": "tsc --noEmit",
    "dev": "yarn build -- watch"
  },
  "repository": {
    "type": "git",
    "url": "git+https://github.com/joe-bell/raam.git"
  },
  "author": "Joe Bell <joe@joebell.co.uk>",
  "license": "ISC",
  "bugs": {
    "url": "https://github.com/joe-bell/raam/issues"
  },
  "homepage": "https://github.com/joe-bell/raam#readme",
  "bundlesize": [
    {
      "path": "./dist/**/*.js",
      "maxSize": "750B"
    }
  ],
  "peerDependencies": {
    "react": ">= 16.8.0",
    "@stitches/core": ">= 0.0.3-canary.4"
  },
  "dependencies": {
    "raam": "^0.2.3"
  },
  "devDependencies": {
    "@stitches/core": "0.0.3-canary.4",
    "@stitches/react": "0.0.3-canary.4",
    "react": "16.13.1",
    "theme-ui": "0.3.1"
  }
}
