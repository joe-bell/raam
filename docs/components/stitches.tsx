import { createFlexbox } from "@raam/stitches";
import { styled, config } from "../styles/stitches.config";

export const Box = styled("div", {});

export const Flexbox = styled("div", {
  ...createFlexbox(config),
  defaultVariants: {
    gap: "4",
    direction: "row",
    wrap: "nowrap",
  },
});
