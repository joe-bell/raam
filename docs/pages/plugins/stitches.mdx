import { Code } from "../../components/code";
import { DemoGrid } from "../../components/demo";
import { Box, Flexbox } from "../../components/stitches";

# Stitches

Flex `gap` for Stitches, without the browser support worries.

Choose your own strategy; use the **util** or **component**.

<DemoGrid variant="stitches" />

## Installation

1. [Install Stitches][stitches]
2. Install the plugin via your preferred package manager:
   ```sh
   npm install -D @raam/stitches
   # or
   yarn add -D @raam/stitches
   ```
3. Choose your strategy/strategies:

   - **Util** – add `flexbox` to your `stitches.config` utils:

     ```js
     // stitches.config.ts
     import { createCss } from "@stitches/react";
     import { flexbox } from "@raam/stitches";

     export const { styled, config, theme } = createCss({
       theme: {
         // …your theme here
       },
       utils: {
         flexbox,
       },
     });
     ```

   - **Component** – create a new styled component–wherever you wish–with the following implementation:

     ```js
     // components/flex.tsx
     import { createFlexbox } from "@raam/stitches";
     import { styled, config } from "../stitches.config";

     export const Flexbox = styled("div", {
       ...createFlexbox(config),
       // Optional: choose your own defaults
       defaultVariants: {
         gap: "4",
         direction: "row",
         wrap: "nowrap",
       },
     });
     ```

## Util

<Code>
  <Box
    css={{
      flexbox: {
        direction: "row",
        wrap: "nowrap",
        gap: "4",
      },
      "> *": {
        width: "$space$10",
        height: "$space$10",
        backgroundColor: "$primary",
        borderRadius: "$md",
      },
    }}
  >
    {Array.from({ length: 4 }).map((_, i, arr) => (
      <div key={i} />
    ))}
  </Box>
</Code>

```tsx
<Box
  css={{
    flexbox: {
      direction: "row",
      wrap: "nowrap",
      gap: "4",
    },
    "> *": {
      width: "$space$10",
      height: "$space$10",
      backgroundColor: "$primary",
      borderRadius: "$md",
    },
  }}
>
  {Array.from({ length: 4 }).map((_, i, arr) => (
    <div key={i} />
  ))}
</Box>
```

## Component

<Code>
  <Flexbox
    css={{
      "> *": {
        width: "$space$10",
        height: "$space$10",
        backgroundColor: "$primary",
        borderRadius: "$md",
      },
    }}
  >
    {Array.from({ length: 4 }).map((_, i, arr) => (
      <div key={i} />
    ))}
  </Flexbox>
</Code>

```tsx
<Flexbox
  direction="row"
  gap="4"
  wrap="nowrap"
  css={{
    "> *": {
      width: "$space$10",
      height: "$space$10",
      backgroundColor: "$primary",
      borderRadius: "$md",
    },
  }}
>
  {Array.from({ length: 4 }).map((_, i, arr) => (
    <div key={i} />
  ))}
</Flexbox>
```

## Recipes

### VStack

<Code>
  <Flexbox
    direction="column"
    gap="4"
    wrap="nowrap"
    css={{
      "> *": {
        height: "$space$10",
        backgroundColor: "$primary",
        borderRadius: "$md",
      },
    }}
  >
    {Array.from({ length: 4 }).map((_, i, arr) => (
      <div key={i} />
    ))}
  </Flexbox>
</Code>

```json
// use as
// 1. options for the `flexbox` util
// 2. props for the `<Flexbox />` component
{
  "direction": "column",
  "gap": "4",
  "wrap": "nowrap"
}
```

### HStack

<Code>
  <Flexbox
    direction="row"
    gap="4"
    wrap="nowrap"
    css={{
      "> *": {
        width: "$space$10",
        height: "$space$10",
        backgroundColor: "$primary",
        borderRadius: "$md",
      },
    }}
  >
    {Array.from({ length: 4 }).map((_, i, arr) => (
      <div key={i} />
    ))}
  </Flexbox>
</Code>

```json
// use as
// 1. options for the `flexbox` util
// 2. props for the `<Flexbox />` component
{
  "direction": "row",
  "gap": "4",
  "wrap": "nowrap"
}
```

### Wrap

<Code>
  <Flexbox
    direction="row"
    gap="5"
    wrap="wrap"
    css={{
      "> *": {
        width: "$space$10",
        height: "$space$10",
        backgroundColor: "$primary",
        borderRadius: "$md",
      },
    }}
  >
    {Array.from({ length: 24 }).map((_, i, arr) => (
      <div key={i} />
    ))}
  </Flexbox>
</Code>

```json
// use as
// 1. options for the `flexbox` util
// 2. props for the `<Flexbox />` component
{
  "direction": "row",
  "gap": "5",
  "wrap": "wrap"
}
```

[stitches]: https://stitches.dev
