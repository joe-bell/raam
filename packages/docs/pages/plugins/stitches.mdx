import { Code } from "../../components/code";
import { DemoGrid } from "../../components/demo";
import { Box, Flexbox } from "../../components/stitches";

# Stitches

Flex `gap` for Stitches, without the browser support worries.

Choose your own strategy; use the **util** or **component**.

<DemoGrid variant="stitches" />

## Installation

1. [Install Stitches][stitches]
2. Install the plugin via your preferred package manager:
   ```sh
   npm install -D @raam/stitches
   # or
   yarn add -D @raam/stitches
   ```
3. Choose your strategy/strategies:

   - **Util** – add `flexbox` to your `stitches.config` utils:

     ```js
     // stitches.config.ts
     import { createCss } from "@stitches/react";
     import { flexbox } from "@raam/stitches";

     export const { styled, config, theme } = createCss({
       theme: {
         // …your theme here
       },
       utils: {
         flexbox,
       },
     });
     ```

   - **Component** – create a new styled component–wherever you wish–with the following implementation:

     ```js
     // components/flex.tsx
     import { createFlexbox } from "@raam/stitches";
     import { styled, config } from "../stitches.config";

     export const Flexbox = styled("div", {
       ...createFlexbox(config),
       // Optional: choose your own defaults
       defaultVariants: {
         gap: "4",
         direction: "row",
         wrap: "nowrap",
       },
     });
     ```

## Util

<Code>
  <Box
    css={{
      flexbox: {
        direction: "row",
        wrap: "nowrap",
        gap: "4",
      },
      "> *": {
        width: "$space$10",
        height: "$space$10",
        backgroundColor: "$purple500",
      },
    }}
  >
    <div />
    <div />
    <div />
    <div />
    <div />
  </Box>
</Code>

```tsx
<Box
  css={{
    flexbox: {
      direction: "row",
      wrap: "nowrap",
      gap: "4",
    },
    "> *": {
      width: "$space$10",
      height: "$space$10",
      backgroundColor: "$purple500",
    },
  }}
>
  <div />
  <div />
  <div />
  <div />
  <div />
</Box>
```

## Component

<Code>
  <Flexbox
    gap="4"
    css={{
      "> *": {
        width: "$space$10",
        height: "$space$10",
        backgroundColor: "$purple500",
      },
    }}
  >
    <div />
    <div />
    <div />
    <div />
    <div />
  </Flexbox>
</Code>

```tsx
<Flexbox
  gap="4"
  css={{
    "> *": {
      width: "$space$10",
      height: "$space$10",
      backgroundColor: "$purple500",
    },
  }}
>
  <div />
  <div />
  <div />
  <div />
  <div />
</Flexbox>
```

## Recipes

[stitches]: https://stitches.dev
