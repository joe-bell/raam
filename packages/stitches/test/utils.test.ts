import * as stitches from "./stitches.config";

describe("flexbox", () => {
  test("should return any gap value", () => {
    const component = stitches.css({
      flexbox: {
        gap: "40px",
      },
    });

    component.toString();
    expect(stitches.toString()).toMatchSnapshot();
  });

  test("should return with themed gap", () => {
    const component = stitches.css({
      flexbox: {
        gap: "$4",
      },
    });

    component.toString();
    expect(stitches.toString()).toMatchSnapshot();
  });

  test("should return responsive themed gap", () => {
    const component = stitches.css({
      flexbox: {
        gap: "$4",
      },
      when: {
        bp1: {
          flexbox: {
            gap: "$6",
          },
        },
      },
    });

    component.toString();
    expect(stitches.toString()).toMatchSnapshot();
  });
});
