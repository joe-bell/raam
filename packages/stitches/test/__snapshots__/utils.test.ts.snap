// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`flexbox should return any gap value 1`] = `":root{--space-1:5px;--space-2:10px;--space-3:15px;--space-4:20px;--space-5:25px;--space-6:35px;--fonts-system:system-ui;}.sxsowz6{--raam-fg-t:40px;--raam-fg-l:40px;--raam-fg-offset:calc(40px * -0.5);--raam-fg:calc(40px * 0.5);margin:var(--raam-fg-offset, initial);box-sizing:border-box;min-width:0;padding-left:0;list-style-type:none;display:flex;}.sxsowz6 > *:first-child{box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, initial);margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, initial);}.sxsowz6 > *:not(:first-child){box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, var(--raam-fg-t, initial));margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, var(--raam-fg-l, initial));}"`;

exports[`flexbox should return responsive themed gap 1`] = `":root{--space-1:5px;--space-2:10px;--space-3:15px;--space-4:20px;--space-5:25px;--space-6:35px;--fonts-system:system-ui;}.sxsowz6{--raam-fg-t:40px;--raam-fg-l:40px;--raam-fg-offset:calc(40px * -0.5);--raam-fg:calc(40px * 0.5);margin:var(--raam-fg-offset, initial);box-sizing:border-box;min-width:0;padding-left:0;list-style-type:none;display:flex;}.sxsowz6 > *:first-child{box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, initial);margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, initial);}.sxsowz6 > *:not(:first-child){box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, var(--raam-fg-t, initial));margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, var(--raam-fg-l, initial));}.sxlap7x{--raam-fg-t:var(--4);--raam-fg-l:var(--4);--raam-fg-offset:calc(var(--4) * -0.5);--raam-fg:calc(var(--4) * 0.5);margin:var(--raam-fg-offset, initial);box-sizing:border-box;min-width:0;padding-left:0;list-style-type:none;display:flex;}.sxlap7x > *:first-child{box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, initial);margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, initial);}.sxlap7x > *:not(:first-child){box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, var(--raam-fg-t, initial));margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, var(--raam-fg-l, initial));}.sxrx0aj{--raam-fg-t:var(--4);--raam-fg-l:var(--4);--raam-fg-offset:calc(var(--4) * -0.5);--raam-fg:calc(var(--4) * 0.5);margin:var(--raam-fg-offset, initial);box-sizing:border-box;min-width:0;padding-left:0;list-style-type:none;display:flex;}.sxrx0aj > *:first-child{box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, initial);margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, initial);}.sxrx0aj > *:not(:first-child){box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, var(--raam-fg-t, initial));margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, var(--raam-fg-l, initial));}@media (min-width: 520px){.sxrx0aj{--raam-fg-t:var(--6);--raam-fg-l:var(--6);--raam-fg-offset:calc(var(--6) * -0.5);--raam-fg:calc(var(--6) * 0.5);margin:var(--raam-fg-offset, initial);box-sizing:border-box;min-width:0;padding-left:0;list-style-type:none;display:flex;}.sxrx0aj > *:first-child{box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, initial);margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, initial);}.sxrx0aj > *:not(:first-child){box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, var(--raam-fg-t, initial));margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, var(--raam-fg-l, initial));}}"`;

exports[`flexbox should return with themed gap 1`] = `":root{--space-1:5px;--space-2:10px;--space-3:15px;--space-4:20px;--space-5:25px;--space-6:35px;--fonts-system:system-ui;}.sxsowz6{--raam-fg-t:40px;--raam-fg-l:40px;--raam-fg-offset:calc(40px * -0.5);--raam-fg:calc(40px * 0.5);margin:var(--raam-fg-offset, initial);box-sizing:border-box;min-width:0;padding-left:0;list-style-type:none;display:flex;}.sxsowz6 > *:first-child{box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, initial);margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, initial);}.sxsowz6 > *:not(:first-child){box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, var(--raam-fg-t, initial));margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, var(--raam-fg-l, initial));}.sxlap7x{--raam-fg-t:var(--4);--raam-fg-l:var(--4);--raam-fg-offset:calc(var(--4) * -0.5);--raam-fg:calc(var(--4) * 0.5);margin:var(--raam-fg-offset, initial);box-sizing:border-box;min-width:0;padding-left:0;list-style-type:none;display:flex;}.sxlap7x > *:first-child{box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, initial);margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, initial);}.sxlap7x > *:not(:first-child){box-sizing:border-box;min-width:0;margin-top:var(--raam-fg, var(--raam-fg-t, initial));margin-right:var(--raam-fg, initial);margin-bottom:var(--raam-fg, initial);margin-left:var(--raam-fg, var(--raam-fg-l, initial));}"`;
